home/api.py

```diff
@@ -3,19 +3,29 @@
 from wagtail.images.api.v2.views import ImagesAPIViewSet
 from wagtail.documents.api.v2.views import DocumentsAPIViewSet
 from .serializers import ContentPageSerializer
+from.models import ContentPageTag
 
 
 class ContentPagesViewSet(PagesAPIViewSet):
     base_serializer_class = ContentPageSerializer
+    known_query_parameters = PagesAPIViewSet.known_query_parameters.union([
+        'tag',
+    ])
+
+    def get_queryset(self):
+        queryset = super(ContentPagesViewSet, self).get_queryset()
+        tag = self.request.query_params.get('tag')
+        if tag is not None:
+            ids = []
+            for t in ContentPageTag.objects.all():
+                if t.tag.name == tag:
+                    ids.append(t.content_object_id)
```

Comment:
Is this a list of Tag objects?
Is that what we're returning or are we returning content pages that have these tags?

Reply:
we are returning the content pages. so we collect the IDs fo the pages that match the tag then filter them against the queryset given by super

Reply:
ah I see. `t.content_object_id` is what I missed. so this is a list of content page ids